ssl/statem/statem_clnt.c

```diff
@@ -1857,37 +1877,37 @@ MSG_PROCESS_RETURN tls_process_server_certificate(SSL *s, PACKET *pkt)
      * set. The *documented* interface remains the same.
      */
     if (s->verify_mode != SSL_VERIFY_NONE && i <= 0) {
+        if (i == -1) {
+            s->rwstate = SSL_RETRY_VERIFY;
+            return WORK_MORE_A;
+        }
```

Comment:
As discussed elsewhere in this review, please double check that `verify_mode` should be required for the new suspension mechanism to work. In my non-expert opinion, the (existing) _unconditional_ verification demands similarly unconditional suspension support.

Reply:
I am pretty sure that the code is fine this way, in the sense that it does not change the uncleanness that was there before regarding the verification mode. 
In client mode we basically only have the two cases `SSL_VERIFY_NONE` and `SSL_VERIFY_PEER` anyway. 
If `SSL_VERIFY_PEER` is used, as will be the case in your application, the feature is enabled as documented.

Reply:
Meanwhile I've made the new feature independent of the verification mode, for two reasons:
* Untangle it from the unclean implementation/documentation regarding `SSL_VERIFY_NONE` vs. `SSL_VERIFY_PEER`
* Generalize its usability, since also the invocation of the callback is independent of the verification mode.